### ISSUE TITLE
Yolo11 training dota2 data set shows loss=nan

### ISSUE THREAD

**Issue by rui-rich** [ultralytics/ultralytics#18466]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

When using yolo11s and yolo11 to train the dota2 dataset, it can be trained normally with imgsz=640, but in order to improve the model effect, I found that using imgsz=1024 would be better. Therefore, I used imgsz=1024 for training. In the early stage of training, yolo11 can train normally. But in the middle, or even in the second round, loss=nan will appear. Why does this happen? Is there any solution?
The training information of yolo11s and yolo11m is as follows：
**************************************************************************************************************
     yolo11s：
     Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     12/120      21.4G      1.268      1.285      1.649        121       1024: 100%|██████████| 4303/4303 [13:09<00:00,  5.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 10093/10093 [07:01<00:00, 23.95it/s]
                   all      50465     716525       0.69       0.47      0.535      0.388
      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     13/120      12.9G        nan        nan        nan         15       1024: 100%|██████████| 4303/4303 [13:12<00:00,  5.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 10093/10093 [05:26<00:00, 30.95it/s]
                   all      50465     716525       0.69       0.47      0.535      0.388
     *******************************************************************************************************
     yolo11m：
     Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      1/120      22.4G       1.25       1.52      1.483          3       1024: 100%|██████████| 7172/7172 [21:48<00:00,  5.48it/s]  
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 8411/8411 [07:17<00:00, 19.24it/s]  
                   all      50465     716525      0.669      0.591      0.602      0.446

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      2/120        12G        nan        nan        nan          0       1024: 100%|██████████| 7172/7172 [19:58<00:00,  5.98it/s] 
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 8411/8411 [04:50<00:00, 28.99it/s]
                   all      50465     716525      0.669      0.591      0.602      0.446

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      3/120      17.4G        nan        nan        nan          1       1024: 100%|██████████| 7172/7172 [20:11<00:00,  5.92it/s]  
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 8411/8411 [04:57<00:00, 28.28it/s]
                   all      50465     716525      0.669      0.591      0.602      0.446

**Comment by UltralyticsAssistant** [comment]
👋 Hello @rui-rich, thank you for your interest in Ultralytics 🚀! We appreciate you bringing up this issue. Below are a few suggestions and resources to help us investigate further and better assist you:

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) so that we can replicate the issue on our end. This would include:

- The exact command or code snippet used to start training.
- Details of your dataset setup (or a subset of your dataset if possible).
- Environment details (operating system, Python version, CUDA version, etc.).
- Any modifications made to the default training configurations.

If you suspect a configuration issue, please verify you are following the recommendations in our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) guide.

## Upgrade 🚀

Ensure you are using the latest version of the `ultralytics` package and that all dependencies are up-to-date. Run the following command to upgrade:

```bash
pip install -U ultralytics
```

This ensures you're working with the latest updates and fixes. For full compatibility, ensure your environment meets the requirements listed in our [pyproject.toml](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml).

## Environments 🌍

YOLO can also be run in the following verified environments. These setups include pre-installed dependencies like [CUDA](https://developer.nvidia.com/cuda) and [PyTorch](https://pytorch.org/):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: Check out our [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/).
- **Amazon Web Services**: See the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/).
- **Docker**: Follow the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>.

## Resources and Community 🧠

Expand your understanding and problem-solving by connecting with the Ultralytics community:
- [Ultralytics Docs](https://docs.ultralytics.com) for Python and CLI usage examples.
- [Discord](https://discord.com/invite/ultralytics) for real-time chat and support 🎧.
- [Discourse](https://community.ultralytics.com) for in-depth discussions.
- [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge and learn from others.

## Status ✅

Kindly verify your training setup against our Continuous Integration tests. If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing:

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

These tests ensure correct operation of YOLO's [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across various platforms.

🎉 An Ultralytics engineer will review and assist further soon. Please share the requested details in the meantime to ensure we can resolve your issue effectively. Thank you!

**Comment by Y-T-G** [comment]
Can you provide the output after running this command in terminal: `yolo checks`?

**Comment by rui-rich** [comment]
> 您能在终端运行此命令后提供输出吗：`yolo checks`？
Yes, here is the output after running the yolo checks command:
Ultralytics 8.3.54 🚀 Python-3.8.20 torch-2.2.2 CUDA:0 (NVIDIA GeForce RTX 4090, 24118MiB)
Setup complete ✅ (64 CPUs, 125.5 GB RAM, 804.1/817.1 GB disk)

OS                  Linux-5.15.0-119-generic-x86_64-with-glibc2.17
Environment         Linux
Python              3.8.20
Install             pip
RAM                 125.53 GB
Disk                804.1/817.1 GB
CPU                 Intel Xeon Gold 6326 2.90GHz
CPU count           64
GPU                 NVIDIA GeForce RTX 4090, 24118MiB
GPU count           8
CUDA                12.1

numpy               ✅ 1.24.3>=1.23.0
numpy               ✅ 1.24.3<2.0.0; sys_platform == "darwin"
matplotlib          ✅ 3.7.5>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 10.4.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.10.1>=1.4.1
torch               ✅ 2.2.2>=1.8.0
torch               ✅ 2.2.2!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.17.2>=0.9.0
tqdm                ✅ 4.66.6>=4.64.0
psutil              ✅ 6.1.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.0.3>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.10>=2.0.0

**Comment by Y-T-G** [comment]
What's your training command?

**Comment by rui-rich** [comment]
> 你的训练指令是什么？

results = model.train(
    data="yolov8_obb/data/dota.yaml",        
    project="yolov8_obb/runs/train",   
    epochs=120,               
    imgsz=1024,
    batch=40,              
    device=[0, 1, 2, 3, 4, 5, 6, 7],
    name="exp",
    optimizer="Adam",      
    lr0=0.001,              
    lrf=0.01,             
    weight_decay=0.0005,   
    # momentum=0.937,         
    warmup_epochs=3.0,     
    amp=True,               
    hsv_h=0.015,          
    hsv_s=0.7,                
    hsv_v=0.4,              
    # degrees=10.0,           
    translate=0.1,      
    scale=0.5,           
    fliplr=0.5,           
    mosaic=0.9,          
    save=True,              
    classes=[0,1,2,3,4,5,6,7,8,9,10,11,12,13,14],
)

**Comment by Y-T-G** [comment]
Remove the optimizer

**Comment by rui-rich** [comment]
> Remove the optimizer

OK, I'll try again. Also, I'd like to ask you, which optimizer should I use? Set it to optimizer=auto, or do not set the optimizer option and just use the default?

**Comment by Y-T-G** [comment]
Do not set

**Comment by visionNoob** [comment]
@rui-rich did you fix the problem?

**Comment by Y-T-G** [comment]
It works either way. But if you are using optimizer=auto then there's no need to pass it as argument.

**Comment by Petros626** [comment]
and why this solves the NaN problem?

**Comment by glenn-jocher** [comment]
The NaN loss during training at larger image sizes (1024 vs 640) often stems from gradient instability. When using `optimizer='auto'`, the system automatically selects optimal learning rates and momentum values based on your dataset and iterations (per our [build_optimizer](https://docs.ultralytics.com/reference/engine/trainer/#ultralytics.engine.trainer.BaseTrainer.build_optimizer) logic), which helps prevent gradient explosions that lead to NaN values. 

For your case with `imgsz=1024` and multi-GPU training, we recommend:
1. Remove all custom optimizer parameters to use auto-tuning
2. Reduce `batch` size if OOM occurs (you're using 40 across 8 GPUs = effective batch 320)
3. Try slightly lowering `lr0` (0.001 → 0.0005) if issues persist

The auto optimizer selection helps maintain stable gradients by adapting to your specific training configuration and dataset characteristics.
